shared/editor/nodes/CodeFence.ts

```diff
@@ -234,6 +284,51 @@ export default class CodeFence extends Node {
     }
   };
 
+  handleDiagramEdit = (event: InputEvent) => {
+    const { view } = this.editor;
+    const { tr } = view.state;
+    const element = event.target;
```

Comment:
```suggestion
    const element = event.currentTarget;
```

I believe you want this to ensure that the event is always coming from the button